docs/design/api-versions.md

```diff
@@ -32,8 +40,169 @@ For each API spec, generated `AssignPropertiesTo*()` method  will read the value
 
 For each Storage spec, the generated `AssignPropertiesTo*()` and `AssignPropertiesFrom*()` methods will copy the `OriginalVersion` property between versions, preserving the original value.
 
-### API Recovery
-
-Into each storage resource variant, we'll inject a function `OriginalGVK(): GroupVersionKind` which will use `OriginalVersion` to create the GVK required.
+## New conversion interfaces
 
-SequenceDiagram TBC.
+We don't want the overhead of doing a complete conversion of each resource if we're only interested in conversion of the associated `Spec` or `Status`. For example, when submitting a request to ARM we need only do a `Spec` conversion; similiarly, when processing a response from ARM, we need only do a `Status` conversion.
+
+We already have the required infrastructure in the form of the `AssignProperties*()` methods, making it simple to implement these new interfaces.
+
+For conversion of `Spec` types:
+
+``` go
+type ConvertibleSpec interface {
+    // ConvertFromSpec populates the current spec from the passed one
+    ConvertFromSpec(source ConvertibleSpec) error
+    
+    // ConvertToSpec populates the passed spec from the current one
+    ConvertToSpec(destination ConvertibleSpec) error
+}
+```
+
+As we have done with implementations of the traditional [`Convertible`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/conversion#Convertible) interface, these will be implemented on Specs that are further from our hub, and will chain towards that hub.
+
+For conversion of `Status` types, a similar approach:
+
+``` go
+type ConvertibleStatus interface {
```

Comment:
Same question about why not just use `Convertible` which the Status types should already implement?

Reply:
And, same answer as for Specs - they don't currently implement `Convertible`, and can't.